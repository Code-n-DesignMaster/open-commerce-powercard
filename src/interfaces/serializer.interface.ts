export interface ISerializer {
  serialize(input: any): any;
  deserialize(input: any): any;
}
