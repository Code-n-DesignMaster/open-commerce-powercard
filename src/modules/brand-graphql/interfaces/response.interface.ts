export interface IResponse {
  readonly success: boolean;
  readonly status: string;
}
