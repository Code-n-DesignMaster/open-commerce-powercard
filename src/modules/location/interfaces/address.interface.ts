import { IGeoLocation } from './geolocation.interface';

export interface IAddress {
  readonly alias: string;
  readonly street1: string;
  readonly street2: string;
  readonly city: string;
  readonly state: string;
  readonly zipCode: string;
  readonly geoLocation?: IGeoLocation;
}
