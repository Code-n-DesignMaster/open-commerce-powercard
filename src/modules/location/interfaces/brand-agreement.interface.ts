export interface IBrandAgreement {
  readonly uuid: string;
  readonly name: string;
  readonly url: string;
  readonly activeAt: Date;
}
