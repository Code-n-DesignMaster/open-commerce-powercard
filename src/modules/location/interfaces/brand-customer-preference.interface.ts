export interface IBrandCustomerPreference {
  readonly uuid: string;
  readonly name: string;
  readonly description: string;
}
