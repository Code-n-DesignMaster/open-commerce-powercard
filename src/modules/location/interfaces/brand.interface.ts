import { ITenant } from './tenant.interface';
import { IBrandAgreement } from './brand-agreement.interface';
import { IBrandCustomerPreference } from './brand-customer-preference.interface';

export interface IBrand {
  name: string;
  tenant: ITenant;
  agreements: IBrandAgreement[];
  preferences: IBrandCustomerPreference[];
}
