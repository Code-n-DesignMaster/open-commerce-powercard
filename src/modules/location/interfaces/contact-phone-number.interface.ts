import { PHONE_TYPE } from '../constants/phone-type.enum';

export interface IContactPhoneNumber {
  uuid: string;
  alias: string;
  phoneNumber: string;
  phoneType: PHONE_TYPE;
}
