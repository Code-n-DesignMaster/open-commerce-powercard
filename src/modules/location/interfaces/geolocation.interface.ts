export interface IGeoLocation {
  readonly latitude: number;
  readonly longitude: number;
}
