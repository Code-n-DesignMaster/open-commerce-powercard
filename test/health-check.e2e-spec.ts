import { INestApplication } from '@nestjs/common/interfaces';
import { Test } from '@nestjs/testing';
import * as request from 'supertest';
import { AppModule } from '../src/modules/app.module';

describe('Health Check (e2e)', () => {
  let app: INestApplication;

  beforeAll(async () => {
    const moduleFixture = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  afterAll(async () => {
    await app.close();
  });

  it('/health-check (GET)', async done => {
    const res = await request(app.getHttpServer())
      .get('/health-check')
      .expect(200);
    expect(res.body).toMatchSnapshot();
    done();
  });
});
