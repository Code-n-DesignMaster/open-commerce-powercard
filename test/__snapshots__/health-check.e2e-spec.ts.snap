// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health Check (e2e) /health-check (GET) 1`] = `
Object {
  "details": Object {},
  "info": Object {},
  "status": "ok",
}
`;
